### DISCUSSION TITLE
设置主机记录与二级域名相关问题.

### DISCUSSION THREAD

**Discussion by Halogly** [jeessy2/ddns-go#1035] [Q&A] [answered]
使用阿里云
现在我有一个顶级域名example.com
出于其他项目需要，我需要给我的exmaple.com设置二级域名，于是我在阿里云的域名解析界面添加域名help.example.com。

在此之前，我已经使用ddns-go在example.com下自动创建了help记录A类型，ip对应公网ip。就是在ddns-go的域名框框里面写上help.example.com

由于我添加了二级域名，原本的域名解析列表：
exmaple.com

变为现在的域名解析列表：
help.example.com
example.com

help.example.com的自带一个记录：@记录A类型，ip对应公网ip。

这里出现了问题，当公网ip改变时，只有example.com的help记录更改了ip地址，而help.example.com的@记录ip地址并没有发生改变，这导致了我的网站解析还是原来的ip地址，进一步导致我的网站无法访问（阿里好像优先选择添加的域名进行解析，也可能不是，个人猜想）

于是我把example.com的help记录删掉，ddns-go报错几次后又重新创建了help记录，help.example.com的ip地址依旧是原来的ip，这使我只能手动更改help.example.com里面的ip地址。

目前想解决的问题是，如何让ddns-go直接修改help.example.com里面的@记录，而不是修改或创建example.com里面的help记录，或者说有什么解决的方法，感谢。

**Comment by jeessy2** [comment]
冒号分割试试，不清楚你是不是多个账号下使用

**Reply by Halogly** [reply]
我只在框框里面写上了help.example.com，并没有把example.com也写进框框里；多账号...如果是指阿里云账号的话，那没有，只有一个账号。

**Reply by jeessy2** [reply]
那不会啊，在检查一下

**Reply by Halogly** [reply]
哦？ddns-go是会一起改的吗，就是单独在阿里云添加了新域名，是help.example.com的二级域名，@记录A类型，上次ip变动时就只改变了example.com的help记录A类型，@记录并没有更改，现在我也只能等到下次ip变动才能知道...

**Reply by Halogly** [reply]
我现在还是有这个问题，我的公网ip变动了，但是，如上所说还是不能都更新ip。

图中单独设置的二级域名ip地址200.176是变更前的地址，而主域名下的记录10.96是变更后的地址，总之就是单独设置了二级域名的不能更新，要自己手动调

<img width="976" alt="42721645f601a9a891bcaaf49ee42f6" src="https://github.com/jeessy2/ddns-go/assets/79710384/e930dad6-42aa-4f7c-9c7e-d592d444aacc">
<img width="976" alt="c573ce3628c009d96699bf8dc411fe2" src="https://github.com/jeessy2/ddns-go/assets/79710384/325a034f-56cd-4fb2-8db5-26706c12690b">
<img width="976" alt="d9a8ee44ed0d7709d153a9ff2086ad0" src="https://github.com/jeessy2/ddns-go/assets/79710384/bdbc9b22-23f1-4a1e-9424-0b4bc3de1781">

**Comment by Halogly** [comment]
新发的看一下，有截图，是一个账号
________________________________
发件人: jeessy2 ***@***.***>
发送时间: 2024年3月8日 23:56
收件人: jeessy2/ddns-go ***@***.***>
抄送: HoneyAlloy ***@***.***>; Author ***@***.***>
主题: Re: [jeessy2/ddns-go] 设置主机记录与二级域名相关问题. (Discussion #1035)


那不会啊，在检查一下

―

**Reply by jeessy2** [reply]
用冒号分割

**Reply by Halogly** [reply]
好的我试试，好久没看了
有个问题是，怎么分割
www.example.com:
api.example.com
---
example.com:www
example.com:api
还是怎么样？

**Reply by jeessy2** [reply, marked as answer]
www:example.com
域名栏下方有说明，多看下吧

**Reply by Halogly** [reply]
我可能用的是旧版本，现在更新就有了，感谢 XD
